kratos/input_output/cad_json_input.h

```diff
@@ -0,0 +1,823 @@
+    typedef BrepCurveOnSurface<ContainerNodeType, ContainerEmbeddedNodeType> BrepCurveOnSurfaceType;
+
+    typedef DenseVector<typename BrepCurveOnSurfaceType::Pointer> BrepCurveOnSurfaceLoopType;
+    typedef DenseVector<DenseVector<typename BrepCurveOnSurfaceType::Pointer>> BrepCurveOnSurfaceLoopArrayType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Constructor with path to input file.
+    CadJsonInput(
+        const std::string& rDataFileName,
+        SizeType EchoLevel = 0)
+        : mEchoLevel(EchoLevel)
+    {
+        mCadJsonParameters = ReadParamatersFile(rDataFileName, EchoLevel);
+    }
+
+    /// Constructor with KratosParameters.
+    CadJsonInput(
+        Parameters CadJsonParameters,
+        SizeType EchoLevel = 0)
+        : mCadJsonParameters(CadJsonParameters)
+        , mEchoLevel(EchoLevel)
+    {
+    }
+
+    /// Destructor.
+    ~CadJsonInput() = default;
+
+    ///@}
+    ///@name Python exposed Functions
+    ///@{
+
+    /// Adds all CAD geometries to the herin provided model_part.
+    void ReadModelPart(ModelPart& rModelPart) override
+    {
+        ReadGeometryModelPart(mCadJsonParameters, rModelPart, mEchoLevel);
+    }
+
+    ///@}
+
+private:
+    ///@name Static Functions
+    ///@{
+
+    /// Allows static access without own memory.
+    static void ReadGeometryModelPart(
+        const Parameters& rCadJsonParameters,
+        ModelPart& rModelPart,
+        SizeType EchoLevel = 0)
+    {
+        KRATOS_ERROR_IF_NOT(rCadJsonParameters.Has("breps"))
+            << "Missing \"breps\" section" << std::endl;
+
+        ReadBreps(rCadJsonParameters["breps"], rModelPart, EchoLevel);
+    }
+
+    ///@}
+    ///@name Read in Brep
+    ///@{
+
+    static void ReadBreps(
+        const Parameters& rParameters,
+        ModelPart& rModelPart,
+        SizeType EchoLevel = 0)
+    {
+        for (IndexType brep_index = 0; brep_index < rParameters.size(); brep_index++)
+        {
+            KRATOS_INFO_IF("ReadBreps", (EchoLevel > 0))
+                << "Reading Brep \"" << GetIdOrName(rParameters[brep_index])
+                << "\" - faces." << std::endl;
+
+            ReadBrepFaces(rParameters[brep_index], rModelPart, EchoLevel);
+        }
+
+        for (IndexType brep_index = 0; brep_index < rParameters.size(); brep_index++)
+        {
+            KRATOS_INFO_IF("ReadBreps", (EchoLevel > 0))
+                << "Reading Brep \"" << GetIdOrName(rParameters[brep_index])
+                << "\" - edges." << std::endl;
+
+            ReadBrepEdges(rParameters[brep_index], rModelPart, EchoLevel);
+        }
+    }
+
+
+    static void ReadBrepFaces(
+        const Parameters& rParameters,
+        ModelPart& rModelPart,
+        SizeType EchoLevel = 0)
+    {
+        if (rParameters.Has("faces"))
+        {
+            ReadBrepSurfaces(rParameters["faces"], rModelPart, EchoLevel);
+        }
+    }
+
+    static void ReadBrepEdges(
+        const Parameters& rParameters,

```

Comment:
```suggestion
        const Parameters Parameters,
```
`Parameters` should be passed by value due to the internal memory management

Reply:
Ok, I didn't know that. Thanks